domain/machine/modelmigration/export.go

```diff
@@ -22,34 +24,97 @@ func RegisterExport(coordinator Coordinator, logger logger.Logger) {
 	})
 }
 
-// ExportService provides a subset of the machine domain
-// service methods needed for machine export.
+// ExportService defines the machine service used to export machines to
+// another controller model to this controller.
 type ExportService interface {
+	// AllMachineNames returns the names of all machines in the model.
 	AllMachineNames(ctx context.Context) ([]coremachine.Name, error)
+	// InstanceID returns the cloud specific instance id for this machine.
+	// If the machine is not provisioned, it returns a NotProvisionedError.
+	InstanceID(ctx context.Context, machineUUID string) (string, error)
+	// HardwareCharacteristics returns the hardware characteristics of the
+	// of the specified machine.
+	HardwareCharacteristics(ctx context.Context, machineUUID string) (*instance.HardwareCharacteristics, error)
+	// GetMachineUUID returns the UUID of a machine identified by its name.
+	// It returns a MachineNotFound if the machine does not exist.
+	GetMachineUUID(ctx context.Context, name coremachine.Name) (string, error)
 }
 
 // exportOperation describes a way to execute a migration for
-// exporting machines.
+// exporting external controllers.
 type exportOperation struct {
 	modelmigration.BaseOperation
 
-	logger  logger.Logger
 	service ExportService
+	logger  logger.Logger
 }
 
 // Name returns the name of this operation.
 func (e *exportOperation) Name() string {
 	return "export machines"
 }
 
-// Setup implements Operation.
 func (e *exportOperation) Setup(scope modelmigration.Scope) error {
-	e.service = service.NewService(
-		state.NewState(scope.ModelDB(), e.logger))
+	e.service = service.NewService(state.NewState(scope.ModelDB(), e.logger))
 	return nil
 }
 
-// Execute the export, adding the machine to the model.
 func (e *exportOperation) Execute(ctx context.Context, model description.Model) error {
+
+	for _, machine := range model.Machines() {
+		// TODO(nvinuesa): We must check if the machine cloud instance is already
+		// set and in that case don't ovewrite anything. This check can be removed
+		// once we fully move machines over to dqlite.
+		if instance := machine.Instance(); instance != nil {
```

Comment:
`instance` clashes with an import, but why not just `if machine.Instance() != nil`?